net/golioth/system_client.c

```diff
@@ -159,12 +147,38 @@ static int golioth_check_credentials(const uint8_t *psk_id, size_t psk_id_len,
 	return err;
 }
 
+static int golioth_check_cert_credentials(void)
+{
+	size_t cred_len = 0;
+	int err = tls_credential_get(CONFIG_GOLIOTH_SYSTEM_CLIENT_CREDENTIALS_TAG,
+				     TLS_CREDENTIAL_SERVER_CERTIFICATE, NULL, &cred_len);
+	if (-ENOENT == err) {
+		LOG_WRN("Certificate authentication configured, but no client certificate found");
+		goto finish;
```

Comment:
```suggestion
		return err;
```

Reply:
I prefer to have a single return point from each function. In a small function like this, it doesn't make much difference, but in larger functions it helps to make the code more readable (subjective) and reduces the likelihood of bugs arising from improper cleanup (objective).